frontend/device/kobo/device.lua

```diff
@@ -829,10 +829,7 @@ function Kobo:initNetworkManager(NetworkMgr)
         self:reconnectOrShowNetworkMenu(complete_callback)
     end
 
-    local net_if = os.getenv("INTERFACE")
-    if not net_if then
-        net_if = "eth0"
-    end
+    local net_if = os.getenv("INTERFACE") or "eth0"
```

Comment:
On a not quite related note (because not applicable to Kobo or whatever), I use this in a local shell script:

```sh
nmcli -t -f device connection show | head -n1
```

Reply:
In a similarly related tangent, the "nice" way of detecting IP assignment would be to *listen* to the netlink socket (`getifaddrs` uses it to pull the data it offers); something that is usually massaged into a slightly more palatable form over dbus via NM ;). (I'm told: I don't use NM :D).